src/components/card/Card.jsx

```diff
@@ -1,30 +1,30 @@
 import "./card.css";
 import { useNavigate } from "react-router-dom";
 import { useAuth, useCart, useWishList } from "../../context";
-export function Card({ product }) {
+export function Card({ product, btnName }) {
   const { user } = useAuth();
   const navigate = useNavigate();
   const { wishList, addToWishList, removeFromWishList } = useWishList();
   const { cart, addToCart } = useCart();
 
-  const isProductInWishList = wishList?.some((p) => p._id === product._id)
+  const isProductInWishList = wishList?.some((item) => item._id === product._id)
     ? true
     : false;
 
-  const isProductInCart = cart?.some((p) => p._id === product._id)
+  const isProductInCart = cart?.some((item) => item._id === product._id)
     ? true
     : false;
 
-  const cartButtonHandler = () => {
-    user
-      ? isProductInCart
-        ? navigate("/cart")
-        : addToCart(product)
-      : navigate("/auth");
-  };
   return (
     <div className="sm-card">
-      <div className="sm-card-img">
+      <div
+        className="sm-card-img"
+        title="Go To Product Page"
+        onClick={(e) => {
+          e.stopPropagation();
```

Comment:
This code is new to me. Could you explain it?

Reply:
It helps with multiple onClick functionality on a single div. To prevent event bubbling we use this to get separate onClick functionalities from the div and buttons used on that div.